app/lib/features/settings/pages/change_password.dart

```diff
@@ -0,0 +1,218 @@
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:logging/logging.dart';
+
+final _log = Logger('a3::room::change_password');
+
+class ChangePasswordPage extends ConsumerStatefulWidget {
+  const ChangePasswordPage({super.key});
+
+  @override
+  ConsumerState<ConsumerStatefulWidget> createState() =>
+      _ChangePasswordPageState();
+}
+
+class _ChangePasswordPageState extends ConsumerState<ChangePasswordPage> {
+  final formKey = GlobalKey<FormState>();
+  final TextEditingController oldPassword = TextEditingController();
+  final TextEditingController newPassword = TextEditingController();
+  final TextEditingController confirmPassword = TextEditingController();
+  bool _oldPasswordVisible = false;
+  bool _newPasswordVisible = false;
+  bool _confirmPasswordVisible = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return WithSidebar(
+      sidebar: const SettingsPage(),
+      child: Scaffold(
+        appBar: _buildAppbar(),
+        body: _buildBody(),
+      ),
+    );
+  }
+
+  AppBar _buildAppbar() {
+    return AppBar(
+      backgroundColor: const AppBarTheme().backgroundColor,
+      elevation: 0.0,
+      title: Text(L10n.of(context).changePassword),
+      centerTitle: true,
+    );
+  }
+
+  Widget _buildBody() {
+    return SingleChildScrollView(
+      child: Center(
+        child: Container(
+          constraints: const BoxConstraints(maxWidth: 500),
+          padding: const EdgeInsets.symmetric(horizontal: 20),
+          child: Form(
+            key: formKey,
+            child: Column(
+              children: [
+                const SizedBox(height: 20),
+                _buildOldPasswordInputField(),
+                const SizedBox(height: 20),
+                _buildNewPasswordInputField(),
+                const SizedBox(height: 20),
+                _buildConfirmPasswordInputField(),
+                const SizedBox(height: 20),
+                _buildChangePasswordButton(),
+                const SizedBox(height: 20),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _buildOldPasswordInputField() {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(L10n.of(context).oldPassword),
+        const SizedBox(height: 10),
+        TextFormField(
+          controller: oldPassword,
+          obscureText: !_oldPasswordVisible,
+          decoration: InputDecoration(
+            hintText: L10n.of(context).hintMessagePassword,
+            suffixIcon: IconButton(
+              icon: Icon(
+                _oldPasswordVisible ? Icons.visibility : Icons.visibility_off,
+              ),
+              onPressed: () {
+                setState(() {
+                  _oldPasswordVisible = !_oldPasswordVisible;
+                });
+              },
+            ),
+          ),
+          inputFormatters: [
+            FilteringTextInputFormatter.deny(RegExp(r'\s')),
+          ],
```

Comment:
I see we also have this denial in the registration and login pages, but I can't find any acceptable spec-reason to have deny them - just this SHOULD https://spec.matrix.org/v1.10/client-server-api/#notes-on-password-management . Which means users _could have_ included spaces in their previous password from different clients and we'd deny them being changed ...?!? At least for the `oldPassword`, we should not enforce this.

```suggestion
```


I think the only/main reason we had that added in the login and registration was that people would add spaces at the end of their password and not realize that and then couldn't login ... But it stands to reason we might want to catch that as a warning on our password fields (like showing that you have capslock on and have a white space at the beginning and/or end) rather than outright denying them being entered. But that is a future thing. I'll make a ticket for fixing them properly (it's a good starter ticket).